src/function.rs

```diff
@@ -76,27 +63,25 @@ impl<'instance> Function<'instance> {
                 wasm_valkind_enum_WASM_I32
                 | wasm_valkind_enum_WASM_FUNCREF
                 | wasm_valkind_enum_WASM_EXTERNREF => {
-                    results.push(WasmValue::decode_to_i32(vec![result[index]]));
+                    results.push(WasmValue::decode_to_i32(result[index..index + 1].to_vec()));
```

Comment:
I'm uncertain about the advantages of using `to_vec()` over `vec!`. Could you please explain?

---

In my view, a potential improvement would be to change the parameter, `binary`, in `decode_to_xxx()` to a reference. This would prevent the need to create a vector just for the function call.